### PR TITLE
`Development`: Add consistency tests for notification placeholders

### PR DESCRIPTION
**Pull request by TimOrtel**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Currently, there are no tests that can verify that developers have not changed the notification placeholders. These are string arrays that are persisted in the database and send to the clients. Removing elements or reordering them can lead to the native clients (Android & iOS) no longer understanding them or misinterpreting them. 

### Description

The following code changes have been made:

1. Put the creation of all placeholder arrays into dedicated methods.
2. Added tests to verify that these placeholder creation methods return correct String arrays based on their parameters.
3. Added a signature test to verify that developers have not accidentally changed the placeholder arrays. Developers changing these methods are now confronted with failing tests. The documentation describes how to handle updated notification placeholders.

### Steps for Testing
This PR is a pure refactoring with added tests, no behavior changes to the server have been made.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

## Summary by CodeRabbit

- **New Features**
	- Enhanced notification system with more descriptive placeholders for various notification types.
- **Refactor**
	- Improved notification creation process with custom annotations and method updates for clarity and efficiency.
- **Tests**
	- Added tests to ensure the reliability and consistency of the notification placeholders and their creation methods.
- **Chores**
	- Updated dependencies to support new testing and notification features.

### CODE REVIEW

**Comment by TimOrtel** [comment]
@coderabbitai resolve

**Comment by TimOrtel** [comment]
@coderabbitai pause

**Review by Strohgelaender — APPROVED** [review]
code

File: src/test/resources/placeholder-signatures.json
Comment (nityanandaz) [inline src/test/resources/placeholder-signatures.json:1]: Note to myself: iOS POI

File: src/test/java/de/tum/in/www1/artemis/domain/notification/NotificationPlaceholderSignatureTest.java
Comment (nityanandaz) [inline src/test/java/de/tum/in/www1/artemis/domain/notification/NotificationPlaceholderSignatureTest.java:28]:
```suggestion
     * If this test fails, you have changed the notification placeholder files. This may have dramatic consequences on the mobile applications (iOS & Android) and the database.
     * Changing the notification placeholders files changes the following things:
     * 1. Other key value pairs are sent to the clients. This may mean that the native clients no longer understand them and notifications are no longer shown to users.
     * 2. The database becomes inconsistent. The placeholders are stored as JSON in the database.
```

File: src/test/java/de/tum/in/www1/artemis/domain/notification/NotificationPlaceholderSignatureTest.java
Comment (nityanandaz) [inline src/test/java/de/tum/in/www1/artemis/domain/notification/NotificationPlaceholderSignatureTest.java:44]:
```suggestion
            // Create a signature for each annotated file.
```

File: src/test/java/de/tum/in/www1/artemis/domain/notification/NotificationPlaceholderSignatureTest.java
Comment (nityanandaz) [inline src/test/java/de/tum/in/www1/artemis/domain/notification/NotificationPlaceholderSignatureTest.java:57]:
```suggestion
            // Signature as JSON.
```

**Review by nityanandaz — COMMENTED** [review]
Code review: functionality looks good

**Review by Strohgelaender — APPROVED** [review]

**Review by nityanandaz — APPROVED** [review]
Re-approve: Resolves comments

File: build.gradle
Comment (krusche) [inline build.gradle:439]: please use the appropriate format

File: src/test/java/de/tum/in/www1/artemis/domain/notification/NotificationPlaceholderSignatureTest.java
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/domain/notification/NotificationPlaceholderSignatureTest.java:59]:
```suggestion
```

**Review by nityanandaz — APPROVED** [review]
I approve on behalf of the iOS application

**Review by Strohgelaender — APPROVED** [review]

**Review by nityanandaz — DISMISSED** [review]
Re-approve after merge develop

**Review by JohannesWt — DISMISSED** [review]
Code looks good

**Review by nityanandaz — APPROVED** [review]
Re-approve after renaming

**Review by julian-christl — APPROVED** [review]
Thank you for considering my request. Code looks good.

**Review by JohannesWt — APPROVED** [review]
Code still looks good :)

**Pull request merged**
